src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -140,6 +140,7 @@ private sealed class RunningInstance
 
             // Calling RunLoopStop multiple times SegFaults so protect the call to it
             private bool _stopping;
+            private object _stopLock = new object();
```

Comment:
I hadn't thought about locking on the instance of the RunningInstance (`this`).

Would I still need the StopLock variable? Couldn't I just do:

```
                lock (this)
                {
                    if (!_stopping && _watcherRunLoop != IntPtr.Zero)
                    {
                        _stopping = true;

                        // Stop the FS event message pump
                        Interop.RunLoop.CFRunLoopStop(_watcherRunLoop);
                    }
                }
```
?